addon/components/validation-error-field.js

```diff
@@ -6,12 +6,18 @@ var ValidationErrorField = Ember.Component.extend({
     attributeBindings: ["array", "className", "model", "field", "validation", "submitted", "delayed", "index"],
     initialize: Ember.on("init", function() {
         var field = this.get("field");
-        Ember.Binding.from("model." + field).to("fieldName").connect(this);
-        Ember.Binding.from("model." + field + "IsPrimed").to("isPrimed").connect(this);
-
         var computedKeys = ["validation", "isPrimed", "submitted", "fieldName"];
+
         if(this.get("array")) {
-            computedKeys.push("array.@each." + field);
+            var fields = field.split(",");
+            fields.forEach(function(f) {
```

Comment:
@williamsbdev finally clicked in my brain and you are correct

1) regular validation support "complex" validation scenarios (meaning we break cache on multiple keys). This was the first commit in the series and I had a test to show it in action
2) array validation support for "complex" validation where we break on multiple using multiple keys

This was a 2 feature commit but one basic "story" in that we now support complex validation across both "single object" and "array" contexts. Let me know if you need more testing around this or if I misunderstood something